starter-code/js/index.js

```diff
@@ -1,52 +1,96 @@
-function deleteItem(e){
-
-}
-
-function getPriceByProduct(itemNode){
-
-}
-
-function updatePriceByProduct(productPrice, index){
+//------------------------------------------------//
+// -- Remove buttons
+//------------------------------------------------//
+
+function deleteItem(e) {
+  // deleteTarget (2 higher: product-wrapper) + parent of deleteTarget (3 higher: wrapper)
+  // e.currentTarget takes the current element of the click event listener (e)
+  var deleteTarget = e.currentTarget.parentNode.parentNode;
+  var parent = e.currentTarget.parentNode.parentNode.parentNode;
+  // remove child (product-wrapper)
+  parent.removeChild(deleteTarget);
+  // recalculate order price after deleting an item
+  getTotalPrice();
+};
 
-}
+ //------------------------------------------------//
+// -- Get total price button
+//------------------------------------------------//
 
 function getTotalPrice() {
+  var orderTotal = 0;
+  for (i = 0; i < document.getElementsByClassName("product-wrapper").length; i++) {
+    // retreive the value 25 with innerHTML (instead of <span>25</span>)
+    var productCost = document.getElementsByClassName("product-cost")[i].innerHTML;
+    // input form has no innerHTML but a value
+    var productQuantity = document.getElementsByClassName("product-quantity")[i].value;
+    // calc total price
+    var totalPrice = productCost * productQuantity;
+    // change price
+    var total = document.getElementsByClassName("product-total")[i];
+    total.innerHTML = totalPrice;
+    orderTotal += totalPrice;
+  };
+  document.getElementById("order-total").innerHTML = orderTotal;
+};
 
-}
-
-function createQuantityInput(){
-
-}
-
-function createDeleteButton(){
-
-}
-
-function createQuantityNode(){
-
-}
-
-function createItemNode(dataType, itemData){
-
-}
-
-function createNewItemRow(itemName, itemUnitPrice){
-
-}
+//------------------------------------------------//
+// -- Create new item
+//------------------------------------------------//
+ 
+function createNewItem() {
+  // retrieve values from input fields
+  var prodName = document.getElementById("input-product-name").value;
+  var prodCost = document.getElementById("input-product-cost").value;
+
+  // error handling
```

Comment:
Great error handling! 

even-though it's not always part of the exercise it's a really good to create this habit 👍